dockstore-webservice/src/main/resources/migrations.1.13.0.xml

```diff
@@ -26,4 +26,14 @@
             <column name="bio" type="text"/>
         </addColumn>
     </changeSet>
+    <changeSet author="ktran" id="convertUserProfileEmptyStringToNull">
+        <update tableName="user_profile">
+            <column name="link"/>
+            <where>link = ''</where>
+        </update>
+        <update tableName="user_profile">
```

Comment:
this migration will ensure this for one point in time, but if we have buggy code or a regression we will get empty strings back in. Is it worth considering a constraint to force this to be either null or non-empty?
Is it ever legitimate for a field to be the empty string (i.e can users put empty strings into their link or bio?)

Reply:
If we add a non-empty constraint for `link`, I think it would make sense to add it to all the columns in that table in case the GitHub API changed in some way. We already saw that the GitHub `name` changed from an empty string to `null` for an unset name.

I tried and it's not legitimate for a field to be an empty string. Also tried a bunch of whitespace and that's not legitimate either.

Reply:
I think the other tricky bit is that this is also used for Google profile, potentially other profiles in the future. 
But I think I do support a follow-up ticket to add constraints since it is confusing programming having to deal with potential empty strings and nulls, both representing the same thing (no data).

Reply:
Created follow up issue https://github.com/dockstore/dockstore/issues/4871